### DISCUSSION TITLE
Popover: InitialFocusRef on input inside a trigger doesn’t work properly

### DISCUSSION THREAD

**Discussion by bielas** [chakra-ui/chakra-ui#6767] [Q&A]
We would like to build simple autocomplete component with an Input element as a popover trigger. Trigger should happen on input click. When popover opens we should be able to type our autocomplete text without focusing on input once again - on popover open, focus should stay on input which at the same time is a trigger. The solution we took is to put initialFocusRef on the input element. However it creates one problem - popover doesnt close on clicking outside the whole component but on clicking again on the trigger - input element. How to achieve that, how to fix that problem?

**Comment by nick-inkeep** [comment]
+1

**Comment by vitalijalbu** [comment]
Anyone got the solution?

**Comment by adk96r** [comment]
Set closeOnBlur={true} on the <Popover> ?
